lib/ffi/cunative/decode_snap.go

```diff
@@ -0,0 +1,525 @@
+	if c.Element == nil {
+		c.Element = new(fr.Element)
+	}
+
+	return c.Element.BigInt(b)
+}
+
+func (c *CursedPoseidonGenRandomnessElement) SetOne() *CursedPoseidonGenRandomnessElement {
+	if c.Element == nil {
+		c.Element = new(fr.Element)
+	}
+
+	c.Element = c.Element.SetOne()
+	return c
+}
+
+func (c *CursedPoseidonGenRandomnessElement) SetZero() *CursedPoseidonGenRandomnessElement {
+	if c.Element == nil {
+		c.Element = new(fr.Element)
+	}
+
+	c.Element = c.Element.SetZero()
+	return c
+}
+
+func (c *CursedPoseidonGenRandomnessElement) Inverse(e *CursedPoseidonGenRandomnessElement) *CursedPoseidonGenRandomnessElement {
+	if c.Element == nil {
+		c.Element = new(fr.Element)
+	}
+
+	c.Element = c.Element.Inverse(e.Element)
+	return c
+}
+
+func (c *CursedPoseidonGenRandomnessElement) Set(e *CursedPoseidonGenRandomnessElement) *CursedPoseidonGenRandomnessElement {
+	if c.Element == nil {
+		c.Element = new(fr.Element)
+	}
+
+	c.Element = c.Element.Set(e.Element)
+	return c
+}
+
+func (c *CursedPoseidonGenRandomnessElement) Square(e *CursedPoseidonGenRandomnessElement) *CursedPoseidonGenRandomnessElement {
+	if c.Element == nil {
+		c.Element = new(fr.Element)
+	}
+
+	c.Element = c.Element.Square(e.Element)
+	return c
+}
+
+func (c *CursedPoseidonGenRandomnessElement) Mul(e2 *CursedPoseidonGenRandomnessElement, e *CursedPoseidonGenRandomnessElement) *CursedPoseidonGenRandomnessElement {
+	if c.Element == nil {
+		c.Element = new(fr.Element)
+	}
+
+	c.Element = c.Element.Mul(e2.Element, e.Element)
+	return c
+}
+
+func (c *CursedPoseidonGenRandomnessElement) Add(e2 *CursedPoseidonGenRandomnessElement, e *CursedPoseidonGenRandomnessElement) *CursedPoseidonGenRandomnessElement {
+	if c.Element == nil {
+		c.Element = new(fr.Element)
+	}
+
+	c.Element = c.Element.Add(e2.Element, e.Element)
+	return c
+}
+
+func (c *CursedPoseidonGenRandomnessElement) Sub(e2 *CursedPoseidonGenRandomnessElement, e *CursedPoseidonGenRandomnessElement) *CursedPoseidonGenRandomnessElement {
+	if c.Element == nil {
+		c.Element = new(fr.Element)
+	}
+
+	c.Element = c.Element.Sub(e2.Element, e.Element)
+	return c
+}
+
+func (c *CursedPoseidonGenRandomnessElement) Cmp(x *CursedPoseidonGenRandomnessElement) int {
+	if c.Element == nil {
+		c.Element = new(fr.Element)
+	}
+
+	return c.Element.Cmp(x.Element)
+}
+
+func (c *CursedPoseidonGenRandomnessElement) SetString(s string) (*CursedPoseidonGenRandomnessElement, error) {
+	if s == "3" {
+		whatTheFuck := "0000000000010000000000000000000000000000000000000000000000000000"
+		dstLE := must.One(hex.DecodeString(whatTheFuck))
+		inverted := make([]byte, len(dstLE))
+		for i := 0; i < len(dstLE); i++ {
+			inverted[i] = dstLE[len(dstLE)-1-i]
+		}
+
+		c.SetBytes(inverted)
+		return c, nil
+	}
+
+	el, err := c.Element.SetString(s)
+	if err != nil {
+		return nil, err
+	}
+
+	c.Element = el
+	return c, nil
+}
+
+var _ poseidon.Element[*CursedPoseidonGenRandomnessElement] = &CursedPoseidonGenRandomnessElement{}
```

Comment:
Cursed?

Reply:
A little.

Gave this a slightly better name